WordPress/src/main/java/org/wordpress/android/ui/reader/repository/ReaderPostRepository.kt

```diff
@@ -1,335 +1,149 @@
-import org.wordpress.android.ui.reader.services.post.ReaderPostServiceStarter.UpdateAction
-import org.wordpress.android.ui.reader.services.post.ReaderPostServiceStarter.UpdateAction.REQUEST_NEWER
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryCommunication.Error.RemoteRequestFailure
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryUseCaseType.FETCH_NUM_POSTS_BY_TAG
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryUseCaseType.FETCH_POSTS_BY_TAG
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryUseCaseType.FETCH_POSTS_BY_TAG_WITH_COUNT
+import org.wordpress.android.ui.reader.repository.ReaderRepositoryUseCaseType.SHOULD_AUTO_UDPATE_TAG
 import org.wordpress.android.util.EventBusWrapper
 import org.wordpress.android.util.NetworkUtilsWrapper
 import org.wordpress.android.viewmodel.ContextProvider
 import org.wordpress.android.viewmodel.Event
 import org.wordpress.android.viewmodel.ReactiveMutableLiveData
 import javax.inject.Inject
 import javax.inject.Named
-import kotlin.coroutines.CoroutineContext
 
-class ReaderPostRepository
-constructor(
-    private val contextProvider: ContextProvider,
+class ReaderPostRepository(
+    private val bgDispatcher: CoroutineDispatcher,
     private val eventBusWrapper: EventBusWrapper,
     private val networkUtilsWrapper: NetworkUtilsWrapper,
-    private val bgDispatcher: CoroutineDispatcher,
-    private val ioDispatcher: CoroutineDispatcher,
-    private val readerTag: ReaderTag
-) : CoroutineScope {
+    private val contextProvider: ContextProvider,
+    private val readerTag: ReaderTag,
+    private val fetchPostsForTagUseCase: FetchPostsForTagUseCase,
+    private val fetchNumPostsForTagUseCase: FetchNumPostsForTagUseCase,
+    private val shouldAutoUpdateTagUseCase: ShouldAutoUpdateTagUseCase,
+    private val fetchPostsForTagWithCountUseCase: FetchPostsForTagWithCountUseCase
+) : BaseReaderRepository(eventBusWrapper,
+        networkUtilsWrapper,
+        contextProvider) {
     private val tagPostsMap: HashMap<ReaderTag, ReaderPostList> = hashMapOf()
+    private val readerRepositoryUseCases:
+            HashMap<ReaderRepositoryUseCaseType, ReaderRepositoryDispatchingUseCase> = hashMapOf()
 
-    // todo: annmarie does the payload need the request with it
     private val _postsForTag = ReactiveMutableLiveData<ReaderPostList>(
             onActive = { onActivePostsForTag() }, onInactive = { onInactivePostsForTag() })
     val postsForTag: ReactiveMutableLiveData<ReaderPostList> = _postsForTag
 
-    // todo: annmarie does the payload need the request with it
-    private val _communicationChannel = MutableLiveData<Event<ReaderRepositoryCommunication>>()
-    val communicationChannel: LiveData<Event<ReaderRepositoryCommunication>> = _communicationChannel
-
     private var isStarted = false
 
-    fun start() {
-        if (isStarted) {
-            return
-        }
+    init {
```

Comment:
Is there any advantage to storing these into a hashmap as oppose to storing them into fields?

Reply:
@malinajirka - I originally had them all in a hashmap passed to the constructor, but I didn't like the clumsiness of casting `as` each time I wanted to use one. Now it only exists for the `stop()` method. I'm not tied to the map and it can go. Next commit.